transcoder/src/subtitles.go

```diff
@@ -53,16 +53,36 @@ outer:
 						sub.IsDefault = true
 					case "forced":
 						sub.IsForced = true
+					case "hi", "sdh", "cc":
+						sub.IsHearingImpaired = true
 					default:
 						lang, err := language.Parse(flag)
 						if err == nil && lang != language.Und {
-							lang := lang.String()
-							sub.Language = &lang
+							langStr := lang.String()
+							sub.Language = &langStr
 						} else {
 							sub.Title = &flag
 						}
 					}
 				}
+
+				// Handle Hindi (hi) collision with Hearing Impaired (hi):
+				// "hi" by itself means a language code, but when combined with other lang flags it means Hearing Impaired.
+				// In case Hindi was not detected before, but "hi" is present, assume it is Hindi.
+				if sub.Language == nil {
+					hiCount := 0
+					for _, flag := range flags {
+						if strings.EqualFold(flag, "hi") {
+							hiCount++
+							langStr := language.Hindi.String()
+							sub.Language = &langStr
+						}
+					}
+					if hiCount == 1 {
+						sub.IsHearingImpaired = false
+					}
```

Comment:
Is there a reason for `EqualFold` here?
I'd rather:

```suggestion
					hiCount := Count(flags, "hi")
					if hiCount > 0 {
						langStr := language.Hindi.String()
						sub.Language = &langStr
					}
					if hiCount == 1 {
						sub.IsHearingImpaired = false
					}
```

with Count being defined in `utils.go` as:

```go
func Counts[S []E, E comparable](s S, e E) int {
	var n int
	for _, v := range s {
		if v == e {
			n++
		}
	}
	return n
}
```

Reply:
The code before is using `strings.ToLower(flag)`. I liked your suggestion, but it will not work if the "HI" flag is uppercase.

Reply:
Good catch, we could switch on `flag` instead of `strings.ToLower(flag)` and change

```go
flags := separator.Split(match[len(base_path):], -1)
```
with

```go
flags_str := strings.ToLower(match[len(base_path):])
flags := separator.Split(flags_str, -1)
```

Reply:
Good, this will work